update 1.X -> 4.X, demo the styles

jakswa opened the pull request:
- big update here, more work to do
- demoing the [`styled(Component)`](https://material-ui.com/styles/basics/) pattern here for the recommended styling pathway moving forward (if we like it... alternate path is doing this update without any of the style-related motivations)
- put this on https://beta.marta.io so you can look at the current ugliness:

![image](https://user-images.githubusercontent.com/137793/66793064-ad9b8b00-eec9-11e9-9590-ed2d0e65ff69.png)

jakswa committed:
update 1.X -> 4.X, demo the styles

- *new* now with fixes and updates for new version. Looking good!

jakswa commented:
still got some oddities on the station view but this is closer

autumncalhoun commented on src/StationListItem/StationListItem.js:
```diff
@@ -11,7 +11,7 @@ class StationListItem {
-      <ListItem divider key={stationName} component={Link} to={link}>
+      <ListItem divider button key={stationName} component={Link} to={link}>
         <ListItemText className="station-list-title" primary={stationName.replace(/ station$/i, '')} />
         <div className="ContainerStationPills">{this.renderPills(arrivals)}</div>
```
I wrote this line previously, all old-skool-like, but I should have done
```
<Box className="ContainerStationPills">{this.renderPills(arrivals)}</Box>
```

autumncalhoun commented on src/StationPills/StationPills.js:
```diff
@@ -1,10 +1,15 @@
 
 const BLANK_CHIP = <Chip className="BlankChip" style={{opacity: 0.5}} label="NO DATA" />;
 
+const MyChip = styled(Chip)({
```
🎉  could we name this `ChipWithCountdown` or `TimerChip` or something?

autumncalhoun commented on public/index.html:
```diff
@@ -64,7 +64,6 @@
       .Pill {
         margin-left: 5px;
         font-weight: 500;
-        font-family: monospace !important;
```
yay!

autumncalhoun committed:
clean up material ui update

jakswa committed:
Merge pull request #27 from jakswa/aw/mat-ui

clean up material ui update

jakswa commented:
merged in @autumncalhoun's cleanup here, and I think it looks good enough to merge. I almost did merge it before noticing the direction font size seems to have taken a hit, and become smaller:

(notice the difference in the N/S/E/W lettering)
![image](https://user-images.githubusercontent.com/137793/69749265-23755200-1118-11ea-8f68-9d20e18a83c0.png)

jakswa committed:
ever-more furious CSS fiddling

Need to just wipe out all the existing styles and come at this fresh,
now that we have things updated. For now I'm just trying to match.

jakswa commented:
Ok I think I have this close, I'm going to get it up on beta.marta.io in a bit:

![image](https://user-images.githubusercontent.com/137793/69754163-35a8bd80-1123-11ea-8885-c48f572dd681.png)

autumncalhoun commented on public/index.html:
```diff
@@ -64,45 +70,48 @@
+        font-size: 1.4rem;
+      }
+
+      .MuiChip-root .Badge.MuiChip-avatar {
```
Sorry about that, my PR didn't change the margin or default height/width of the circle but I thought I fixed the small font. Anyway, it looks good now on my local.

jakswa replied:
fwiw I don't think any of that was your doing. I think it was the version upgrade's fault.
